--- a/3rdparty/fltk/FL/Fl_Color_Chooser.H
+++ b/3rdparty/fltk/FL/Fl_Color_Chooser.H
@@ -183,6 +183,9 @@ public:
 FL_EXPORT int fl_color_chooser(const char* name, double& r, double& g, double& b, int m=-1);
 FL_EXPORT int fl_color_chooser(const char* name, uchar& r, uchar& g, uchar& b, int m=-1);
 
+// 2016/12/10  djcj <djcj@gmx.de>
+FL_EXPORT int fl_color_chooser_nodeco(const char* name, double& r, double& g, double& b, int m=-1);
+
 #endif
 
 //
--- a/3rdparty/fltk/src/Fl_Color_Chooser.cxx
+++ b/3rdparty/fltk/src/Fl_Color_Chooser.cxx
@@ -617,6 +617,39 @@ int fl_color_chooser(const char* name, uchar& r, uchar& g, uchar& b, int cmode)
   return 0;
 }
 
+// 2016/12/10  djcj <djcj@gmx.de>
+int fl_color_chooser_nodeco(const char* name, double& r, double& g, double& b, int cmode) {
+  int ret = 0;
+  Fl_Window window(215,200,name);
+  window.callback(cc_cancel_cb,&ret);
+  Fl_Color_Chooser chooser(10, 10, 195, 115);
+  ColorChip ok_color(10, 130, 95, 25);
+  Fl_Return_Button ok_button(10, 165, 95, 25, fl_ok);
+  ok_button.callback(cc_ok_cb,&ret);
+  ColorChip cancel_color(110, 130, 95, 25);
+  cancel_color.r = uchar(255*r+.5); ok_color.r = cancel_color.r;
+  ok_color.g = cancel_color.g = uchar(255*g+.5);
+  ok_color.b = cancel_color.b = uchar(255*b+.5);
+  Fl_Button cancel_button(110, 165, 95, 25, fl_cancel);
+  cancel_button.callback(cc_cancel_cb,&ret);
+  window.resizable(chooser);
+  chooser.rgb(r,g,b);
+  chooser.callback(chooser_cb, &ok_color);
+  if (cmode!=-1) chooser.mode(cmode);
+  window.end();
+  window.set_modal();
+  window.hotspot(window);
+  window.show();
+  window.border(0);  // 2016/12/10  djcj <djcj@gmx.de>
+  while (window.shown()) Fl::wait();
+  if (ret) { // ok_button or Enter
+    r = chooser.r();
+    g = chooser.g();
+    b = chooser.b();
+  }
+  return ret;
+}
+
 /** @} */
 //
 // End of "$Id: Fl_Color_Chooser.cxx 10234 2014-08-21 12:18:32Z cand $".
