--- a/FL/fl_ask.H
+++ b/FL/fl_ask.H
@@ -69,6 +69,8 @@
 FL_EXPORT void fl_message_title(const char *title);
 FL_EXPORT void fl_message_title_default(const char *title);
 
+FL_EXPORT void fl_message_size(int w, int h);
+
 // pointers you can use to change FLTK to a foreign language:
 extern FL_EXPORT const char* fl_no;
 extern FL_EXPORT const char* fl_yes;
--- a/src/fl_ask.cxx
+++ b/src/fl_ask.cxx
@@ -58,6 +58,9 @@
 extern "C" void NSBeep(void);
 #endif
 
+static int override_w = 0;
+static int override_h = 0;
+
 static char avoidRecursion = 0;
 
 // Sets the global return value (ret_val) and closes the window.
@@ -146,6 +149,9 @@
   if (message_h < 30)
     message_h = 30;
 
+  if (override_w > 60) message_w = override_w;
+  if (override_h > 30) message_h = override_h;
+
   fl_font(button[0]->labelfont(), button[0]->labelsize());
 
   memset(button_w, 0, sizeof(button_w));
@@ -588,6 +594,8 @@
   message_form->copy_label(title);
 }
 
+void fl_message_size(int w, int h) { override_w = w; override_h = h; }
+
 /** Sets the default title of the dialog window used in many common dialogs.
 
     This window \p title will be used in all subsequent calls of one of the
