--- a/src/fl_ask.cxx
+++ b/src/fl_ask.cxx
@@ -71,6 +71,10 @@ static void button_cb(Fl_Widget *, long val) {
   message_form->hide();
 }
 
+void ask_window_cb(Fl_Widget*) {
+  exit(1);
+}
+
 static Fl_Window *makeform() {
  if (message_form) {
    return message_form;
@@ -81,7 +85,7 @@ static Fl_Window *makeform() {
  Fl_Group::current(0);
  // create a new top level window
  Fl_Window *w = message_form = new Fl_Window(410,103);
-  message_form->callback(button_cb);
+  message_form->callback(ask_window_cb);
  // w->clear_border();
  // w->box(FL_UP_BOX);
  (message = new Fl_Box(60, 25, 340, 20))
@@ -489,7 +493,11 @@ static const char* input_innards(const char* fmt, va_list ap,
   int r = innards(fmt, ap, fl_cancel, fl_ok, 0);
   input->hide();
   message->position(60,25);
-  return r ? input->value() : 0;
+  if (r) {
+    return input->value();
+  } else {
+    exit(1);
+  }
 }
 
 /** Shows an input dialog displaying the \p fmt message
